diff -ur cdbs-0.4.74.old/doc/Makefile cdbs-0.4.74/doc/Makefile
--- cdbs-0.4.74.old/doc/Makefile	2010-03-25 16:46:11.695146653 +0100
+++ cdbs-0.4.74/doc/Makefile	2010-03-25 16:47:19.499147438 +0100
@@ -12,11 +12,11 @@
 
 pdf: cdbs-doc.pdf
 %.pdf: %.xml xmllint-stamp-% $(graphics:%=%.pdf)
-	dblatex -b pdftex -T db2latex $<
+	docbook2latex -b pdftex -T db2latex $<
 
 ps: cdbs-doc.ps
 %.ps: %.xml xmllint-stamp-% $(graphics:%=%.eps)
-	dblatex -T db2latex -t ps -f eps $<
+	docbook2latex -T db2latex -t ps -f eps $<
 
 txt: cdbs-doc.txt
 %.txt: %.html
