Index: broadcom-sta-5.100.82.112/amd64/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta-5.100.82.112.orig/amd64/src/wl/sys/wl_linux.c	2011-11-12 12:29:08.000000000 +0400
+++ broadcom-sta-5.100.82.112/amd64/src/wl/sys/wl_linux.c	2011-11-12 12:29:08.000000000 +0400
@@ -258,7 +258,7 @@
 #define quote_str(s) to_str(s)
 
 #ifndef BRCM_WLAN_IFNAME
-#define BRCM_WLAN_IFNAME eth%d
+#define BRCM_WLAN_IFNAME wlan%d
 #endif
 
 static char name[IFNAMSIZ] = quote_str(BRCM_WLAN_IFNAME);
