--- ufraw-0.19.2.old/configure.ac	2013-03-25 02:40:46.000000000 +0000
+++ ufraw-0.19.2/configure.ac	2013-04-12 12:21:55.964670888 +0100
@@ -169,57 +169,120 @@
 AC_SUBST(CINEPAINT_PROGRAMPLUGINDIR)
 
 # Check for zlib.
-AC_CHECK_LIB(z, deflate)
+AC_ARG_WITH([zlib],
+  [AS_HELP_STRING([--with-zlib],
+  [build zlib support <:@default=check@:>@])],
+  [],
+  [with_zlib=check])
+
+AS_IF([test "x$with_zlib" != "xno"],
+  [AC_CHECK_LIB([z], [deflate])])
 have_zlib=${ac_cv_lib_z_deflate:-no}
 
 # Check for libbz2.
-AC_CHECK_LIB(bz2, BZ2_bzReadOpen)
+AC_ARG_WITH([bzip2],
+  [AS_HELP_STRING([--with-bzip2], 
+  [build bzip2 support <:@default=check@:>@])],
+  [],
+  [with_bzip2=check]) 
+
+AS_IF([test "x$with_bzip2" != "xno"],
+  [AC_CHECK_LIB([bz2], [BZ2_bzReadOpen])])
 have_libbz2=${ac_cv_lib_bz2_BZ2_bzReadOpen:-no}
 
 # Check for jpeg headers and library.
-AC_CHECK_HEADER(jpeglib.h,
-  AC_CHECK_LIB(jpeg, jpeg_CreateCompress))
+AC_ARG_WITH([jpeg],
+  [AS_HELP_STRING([--with-jpeg],
+  [build jpeg support <:@default=check@:>@])],
+  [],
+  [with_jpeg=check])
+
+AS_IF([test "x$with_jpeg" != "xno"],
+  [AC_CHECK_HEADER([jpeglib.h],
+    [AC_CHECK_LIB([jpeg], [jpeg_CreateCompress])])])
 have_jpeg=${ac_cv_lib_jpeg_jpeg_CreateCompress:-no}
 
 # Check for libjasper.
-AC_CHECK_LIB(jasper, jas_image_decode)
+AC_ARG_WITH([jasper],
+  [AS_HELP_STRING([--with-jasper],
+  [build jasper support <:@default=check@:>@])],
+  [],
+  [with_jasper=check])
+
+AS_IF([test "x$with_jasper" != "xno"],
+  [AC_CHECK_LIB([jasper], [jas_image_decode])])
 have_jasper=${ac_cv_lib_jasper_jas_image_decode:-no}
 
 # Check for tiff headers and library.
-PKG_CHECK_MODULES(LIBTIFF, libtiff-4,
-  [ have_tiff=yes
-    AC_DEFINE(HAVE_LIBTIFF, 1, have the tiff library) ],
-  [ have_tiff=no
-    AC_MSG_RESULT($LIBTIFF_PKG_ERRORS) ] )
-if test $have_tiff = no; then
-  AC_CHECK_HEADER(tiffio.h,
-    AC_CHECK_LIB(tiff, TIFFSetErrorHandler))
-  have_tiff=${ac_cv_lib_tiff_TIFFSetErrorHandler:-no}
-fi
-
-PKG_CHECK_MODULES(LIBPNG, libpng >= 1.2,
-  [ have_png=yes
-    AC_DEFINE(HAVE_LIBPNG, 1, have the png library) ],
-  [ have_png=no
-    AC_MSG_RESULT($LIBPNG_PKG_ERRORS) ] )
-
-PKG_CHECK_MODULES(CFITSIO, cfitsio,
-  [ have_cfitsio=yes
-    AC_DEFINE(HAVE_LIBCFITSIO, 1, have the cfitsio library) ],
-  [ have_cfitsio=no
-    AC_MSG_RESULT($CFITSIO_PKG_ERRORS) ] )
-
-PKG_CHECK_MODULES(EXIV2, exiv2 >= 0.20,
-  [ have_exiv2=yes
-    AC_DEFINE(HAVE_EXIV2, 1, have the exiv2 library) ],
-  [ have_exiv2=no
-    AC_MSG_RESULT($EXIV2_PKG_ERRORS) ] )
-
-PKG_CHECK_MODULES(LENSFUN, lensfun >= 0.2.5,
-  [ have_lensfun=yes
-    AC_DEFINE(HAVE_LENSFUN, 1, have the lensfun library) ],
-  [ have_lensfun=no
-    AC_MSG_RESULT($LENSFUN_PKG_ERRORS) ] )
+AC_ARG_WITH([libtiff],
+  [AS_HELP_STRING([--with-libtiff],
+  [build libtiff support <:@default=check@:>@])],
+  [],
+  [with_libtiff=check])
+
+have_libtiff=no
+AS_IF([test "x$with_libtiff" != xno],
+  [PKG_CHECK_MODULES(LIBTIFF, libtiff-4,
+    [ have_libtiff=yes
+      AC_DEFINE(HAVE_LIBTIFF, 1, have the tiff library) ],
+    [ have_libtiff=no
+      AC_MSG_RESULT($LIBTIFF_PKG_ERRORS)])])
+
+AC_ARG_WITH([libpng],
+  [AS_HELP_STRING([--with-libpng],
+  [build libpng support <:@default=check@:>@])],
+  [],
+  [with_libpng=check])
+
+have_png=no
+AS_IF([test "x$with_libpng" != xno],
+  [PKG_CHECK_MODULES(LIBPNG, libpng >= 1.2,
+    [ have_png=yes
+      AC_DEFINE(HAVE_LIBPNG, 1, have the png library) ],
+    [ have_png=no
+      AC_MSG_RESULT($LIBPNG_PKG_ERRORS)])])
+
+AC_ARG_WITH([cfitsio],
+  [AS_HELP_STRING([--with-cfitsio],
+  [build fits support @<:@default=check@:>@])],
+  [],
+  [with_cfitsio=check])
+
+have_cfitsio=no
+AS_IF([test "x$with_cfitsio" != xno],
+  [PKG_CHECK_MODULES(CFITSIO, cfitsio,
+    [ have_cfitsio=yes
+      AC_DEFINE(HAVE_LIBCFITSIO, 1, have the cfitsio library) ],
+    [ have_cfitsio=no
+      AC_MSG_RESULT($CFITSIO_PKG_ERRORS)])])
+
+AC_ARG_WITH([exiv2],
+  [AS_HELP_STRING([--with-exiv2],
+  [build exiv2 support @<:@default=check@:>@])],
+  [],
+  [with_exiv2=check])
+
+have_exiv2=no
+AS_IF([test "x$with_exiv2" != xno],
+  [PKG_CHECK_MODULES(EXIV2, exiv2 >= 0.20,
+    [ have_exiv2=yes
+      AC_DEFINE(HAVE_EXIV2, 1, have the exiv2 library) ],
+    [ have_exiv2=no
+      AC_MSG_RESULT($EXIV2_PKG_ERRORS)])])
+
+AC_ARG_WITH([lensfun],
+  [AS_HELP_STRING([--with-lensfun],
+  [build lensfun support @<:@default=check@:>@])],
+  [],
+  [with_lensfun=check])
+
+have_lensfun=no
+AS_IF([test "x$with_lensfun" != xno],
+  [PKG_CHECK_MODULES(LENSFUN, lensfun >= 0.2.5,
+    [ have_lensfun=yes
+      AC_DEFINE(HAVE_LENSFUN, 1, have the lensfun library) ],
+    [ have_lensfun=no
+      AC_MSG_RESULT($LENSFUN_PKG_ERRORS)])])
 
 # Make sure MSGFMT_OPTS is defined for all combinations of autoconf and glib.
 MSGFMT_OPTS=
