diff -urN gconf-editor-2.28.0.old/src/gconf-editor-window.c gconf-editor-2.28.0/src/gconf-editor-window.c
--- gconf-editor-2.28.0.old/src/gconf-editor-window.c	2009-09-21 13:11:36.000000000 +0200
+++ gconf-editor-2.28.0/src/gconf-editor-window.c	2010-05-10 23:23:56.005460923 +0200
@@ -254,7 +254,11 @@
 		gconf_client_set_list (window->client,
 				       "/apps/gconf-editor/recents", GCONF_VALUE_STRING, list, NULL);
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+		if (gtk_widget_get_visible (GTK_WIDGET (window->output_window)) &&
+#else
 		if (GTK_WIDGET_VISIBLE (GTK_WIDGET (window->output_window)) &&
+#endif
 	    				window->output_window_type == GCONF_EDITOR_WINDOW_OUTPUT_WINDOW_RECENTS) 
 			gedit_output_window_prepend_line (GEDIT_OUTPUT_WINDOW (window->output_window),
 							  (gchar*) key, TRUE);
@@ -613,7 +617,11 @@
   GConfEditorWindow *gconfwindow = GCONF_EDITOR_WINDOW (data);
   GdkCursor *cursor;
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (!gtk_widget_get_realized (gconfwindow))
+#else
   if (!GTK_WIDGET_REALIZED (gconfwindow))
+#endif
     return FALSE;
 
   cursor = gdk_cursor_new (GDK_WATCH);
@@ -633,7 +641,11 @@
 {
   GConfEditorWindow *gconfwindow = GCONF_EDITOR_WINDOW (data);
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (!gtk_widget_get_realized (gconfwindow))
+#else
   if (!GTK_WIDGET_REALIZED (gconfwindow))
+#endif
     return;
 
   gdk_window_set_cursor (GTK_WIDGET (gconfwindow)->window, NULL);
