diff -urN gnome-terminal-2.28.2.old/src/terminal-screen.c gnome-terminal-2.28.2/src/terminal-screen.c
--- gnome-terminal-2.28.2.old/src/terminal-screen.c	2009-12-14 21:22:36.000000000 +0100
+++ gnome-terminal-2.28.2/src/terminal-screen.c	2010-05-10 16:39:30.975634673 +0200
@@ -263,7 +263,11 @@
       g_return_if_fail (GTK_IS_NOTEBOOK (widget->parent));
 
       toplevel = gtk_widget_get_toplevel (widget);
+#if GTK_CHECK_VERSION (2, 19, 7)
+      g_return_if_fail (gtk_widget_is_toplevel (toplevel));
+#else
       g_return_if_fail (GTK_WIDGET_TOPLEVEL (toplevel));
+#endif
 
       priv->window = TERMINAL_WINDOW (toplevel);
     }
@@ -346,7 +350,11 @@
 
   update_color_scheme (screen);
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (gtk_widget_get_realized (widget))
+#else
   if (GTK_WIDGET_REALIZED (widget))
+#endif
     terminal_screen_change_font (screen);
 }
 
@@ -946,7 +954,11 @@
       terminal_screen_cook_icon_title (screen);
     }
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (gtk_widget_get_realized (screen) &&
+#else
   if (GTK_WIDGET_REALIZED (screen) &&
+#endif
       (!prop_name ||
        prop_name == I_(TERMINAL_PROFILE_USE_SYSTEM_FONT) ||
        prop_name == I_(TERMINAL_PROFILE_FONT)))
@@ -1135,7 +1147,11 @@
 {
   TerminalScreenPrivate *priv = screen->priv;
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (!gtk_widget_get_realized (screen))
+#else
   if (!GTK_WIDGET_REALIZED (screen))
+#endif
     return;
 
   if (!terminal_profile_get_property_boolean (priv->profile, TERMINAL_PROFILE_USE_SYSTEM_FONT))
@@ -1370,7 +1386,11 @@
   guint i;
 
   window = gtk_widget_get_toplevel (term);
+#if GTK_CHECK_VERSION (2, 19, 7)
+  g_assert (window != NULL && gtk_widget_is_toplevel (window));
+#else
   g_assert (window != NULL && GTK_WIDGET_TOPLEVEL (window));
+#endif
 
   env_table = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
 
@@ -1864,7 +1884,11 @@
   
   priv->font_scale = factor;
   
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (gtk_widget_get_realized (screen))
+#else
   if (GTK_WIDGET_REALIZED (screen))
+#endif
     {
       /* Update the font */
       terminal_screen_change_font (screen);
diff -urN gnome-terminal-2.28.2.old/src/terminal-window.c gnome-terminal-2.28.2/src/terminal-window.c
--- gnome-terminal-2.28.2.old/src/terminal-window.c	2009-12-14 21:22:36.000000000 +0100
+++ gnome-terminal-2.28.2/src/terminal-window.c	2010-05-10 16:54:14.111147863 +0200
@@ -305,7 +305,11 @@
       tab = gtk_notebook_get_tab_label (nb, page);
       g_return_val_if_fail (tab != NULL, -1);
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+      if (!gtk_widget_get_mapped (GTK_WIDGET (tab)))
+#else
       if (!GTK_WIDGET_MAPPED (GTK_WIDGET (tab)))
+#endif
         {
           page_num++;
           continue;
@@ -356,7 +360,11 @@
   gdk_screen_get_monitor_geometry (screen, monitor_num, &monitor);
 
   gdk_window_get_origin (widget->window, x, y);
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (gtk_widget_get_has_window (widget))
+#else
   if (GTK_WIDGET_NO_WINDOW (widget))
+#endif
     {
       *x += widget->allocation.x;
       *y += widget->allocation.y;
@@ -984,7 +992,11 @@
 
   /* Don't do anything if we're maximised or fullscreened */
   // FIXME: realized && ... instead? 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (!gtk_widget_get_realized (widget) ||
+#else
   if (!GTK_WIDGET_REALIZED (widget) ||
+#endif
       (gdk_window_get_state (widget->window) & (GDK_WINDOW_STATE_MAXIMIZED | GDK_WINDOW_STATE_FULLSCREEN)) != 0)
     return;
 
@@ -1227,7 +1239,11 @@
   gboolean can_paste, can_paste_uris, show_link, show_email_link, show_call_link, show_input_method_menu;
   int n_pages;
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (!gtk_widget_get_realized (info ->screen))
+#else
   if (!GTK_WIDGET_REALIZED (info->screen))
+#endif
     {
       terminal_screen_popup_info_unref (info);
       return;
@@ -1550,7 +1566,11 @@
 
   composited = gdk_screen_is_composited (screen);
   if ((composited != priv->have_argb_visual) &&
+#if GTK_CHECK_VERSION (2, 19, 7)
+      gtk_widget_get_realized (window))
+#else
       GTK_WIDGET_REALIZED (window))
+#endif
     {
       GtkWidget *widget = GTK_WIDGET (window);
       guint32 user_time;
@@ -2210,7 +2230,11 @@
 terminal_window_set_is_restored (TerminalWindow *window)
 {
   g_return_if_fail (TERMINAL_IS_WINDOW (window));
+#if GTK_CHECK_VERSION (2, 19, 7)
+  g_return_if_fail (!gtk_widget_get_mapped (window));
+#else
   g_return_if_fail (!GTK_WIDGET_MAPPED (window));
+#endif
 
   window->priv->clear_demands_attention = TRUE;
 }
@@ -2222,7 +2246,11 @@
 {
   TerminalWindowPrivate *priv = window->priv;
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (!gtk_widget_get_realized (window))
+#else
   if (!GTK_WIDGET_REALIZED (window))
+#endif
     return;
 
   if (screen != priv->active_screen)
@@ -2269,7 +2297,11 @@
 {
   TerminalWindowPrivate *priv = window->priv;
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (!gtk_widget_get_realized (window))
+#else
   if (!GTK_WIDGET_REALIZED (window))
+#endif
     return;
 
   /* No need to restore the title if we never set an icon title */
@@ -2305,7 +2337,11 @@
   TerminalScreen *screen;
 
   toplevel = gtk_widget_get_toplevel (screen_container);
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (!gtk_widget_is_toplevel (toplevel) || !TERMINAL_IS_WINDOW (toplevel))
+#else
   if (!GTK_WIDGET_TOPLEVEL (toplevel) || !TERMINAL_IS_WINDOW (toplevel))
+#endif
     return;
 
   window = TERMINAL_WINDOW (toplevel);
@@ -2328,7 +2364,11 @@
   GtkWidget *screen_container, *tab_label;
 
   old_window = gtk_widget_get_toplevel (GTK_WIDGET (screen));
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (gtk_widget_is_toplevel (old_window) &&
+#else
   if (GTK_WIDGET_TOPLEVEL (old_window) &&
+#endif
       TERMINAL_IS_WINDOW (old_window) &&
       TERMINAL_WINDOW (old_window)== window)
     return;  
@@ -2540,7 +2580,11 @@
                          window,
                          grid_width, grid_height, force_grid_width, force_grid_height, w, h);
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (even_if_mapped && gtk_widget_get_mapped (app)) {
+#else
   if (even_if_mapped && GTK_WIDGET_MAPPED (app)) {
+#endif
     gtk_window_resize (GTK_WINDOW (app), w, h);
   }
   else {
@@ -2577,7 +2621,11 @@
 
   profile = terminal_screen_get_profile (screen);
 
+#if GTK_CHECK_VERSION (2, 19, 7)
+  if (!gtk_widget_get_realized (widget))
+#else
   if (!GTK_WIDGET_REALIZED (widget))
+#endif
     gtk_widget_realize (widget); /* we need this for the char width */
 
   priv->active_screen = screen;
@@ -3309,7 +3357,11 @@
 view_fullscreen_toggled_callback (GtkToggleAction *action,
                                   TerminalWindow *window)
 {
+#if GTK_CHECK_VERSION (2, 19, 7)
+  g_return_if_fail (gtk_widget_get_realized (window));
+#else
   g_return_if_fail (GTK_WIDGET_REALIZED (window));
+#endif
 
   if (gtk_toggle_action_get_active (action))
     gtk_window_fullscreen (GTK_WINDOW (window));
