diff -urN shotwell-0.11.0.old//Makefile shotwell-0.11.0/Makefile
--- shotwell-0.11.0.old//Makefile	2011-08-23 20:53:10.000000000 +0200
+++ shotwell-0.11.0/Makefile	2011-08-31 11:51:51.909120001 +0200
@@ -545,7 +545,6 @@
 	mkdir -p $(DESTDIR)$(PREFIX)/share/glib-2.0/schemas
 	$(INSTALL_DATA) misc/org.yorba.shotwell.gschema.xml $(DESTDIR)$(PREFIX)/share/glib-2.0/schemas
 	$(INSTALL_DATA) misc/org.yorba.shotwell-extras.gschema.xml $(DESTDIR)$(PREFIX)/share/glib-2.0/schemas
-	glib-compile-schemas $(DESTDIR)$(PREFIX)/share/glib-2.0/schemas
 	mkdir -p $(DESTDIR)/usr/share/GConf/gsettings
 	$(INSTALL_DATA) misc/shotwell.convert $(DESTDIR)/usr/share/GConf/gsettings
 ifndef DISABLE_ICON_UPDATE
@@ -636,7 +635,6 @@
 endif
 	rm -f $(DESTDIR)$(PREFIX)/share/glib-2.0/schemas/org.yorba.shotwell.gschema.xml
 	rm -f $(DESTDIR)$(PREFIX)/share/glib-2.0/schemas/org.yorba.shotwell-extras.gschema.xml
-	glib-compile-schemas $(DESTDIR)$(PREFIX)/share/glib-2.0/schemas
 	rm -f $(DESTDIR)/usr/share/GConf/gsettings/shotwell.convert
 
 $(PC_FILE): $(PC_INPUT) $(MAKE_FILES)
